test/integration/token-discovery.bats

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function token_cleanup() {
+	curl -X DELETE https://discovery-stage.hub.docker.com/v1/clusters/$1
+}
+
+function teardown() {
+	swarm_join_cleanup
+	swarm_manage_cleanup
+	stop_docker
+}
+
+@test "token discovery should be working properly" {
+	start_docker 2
+
+	TOKEN=$(swarm create)
+	[ "$status" -eq 0 ]
+	[[ ${TOKEN} =~ ^[0-9a-f]{32}$ ]]
+
+	swarm_manage token://$TOKEN
+	swarm_join   token://$TOKEN
+
+	retry 10 1 docker_swarm info && [[ "${lines[3]}" == *"Nodes: 2"* ]]
```

Comment:
Oh, I see the problem - basically `docker list` returns 2 items faster than what swarm sees, right?

In that case, what about changing the test in swarm_join from:
`[ `swarm list "$@" | wc -l` -eq $i ]`
to something that runs docker_swarm info. Maybe:
[ -n `docker_swarm info | grep "Nodes: $i" ]